### PR TITLE
Init dashes

### PR DESCRIPTION
**Pull request by fortmarek**
Resolves https://github.com/tuist/tuist/issues/1741

### Short description 📝

Target with dashes is unsupported, we should support them.

### Solution 📦

To fix it we are now camelizing the name of the initialized project with already-defined `camelized` function

### Implementation 👩‍💻👨‍💻

> Detail in a checklist the steps that you took to implement the PR.

- [x] Implement
- [x] Add acceptance test to catch this
- [x] Add changelog

### CODE REVIEW

File: Templates/AppProject.stencil [hunk 1]
```diff
@@ -22,4 +22,4 @@ import ProjectDescriptionHelpers
 // Creates our project using a helper function defined in ProjectDescriptionHelpers
 let project = Project.app(name: "{{ name }}",
                           platform: .{{ platform }},
-                          additionalTargets: ["{{name}}Kit", "{{name}}UI"])
+                          additionalTargets: ["{{name }}Kit", "{{name}}UI"])
```
Comment (laxmorek) [inline Templates/AppProject.stencil:25]: Should not be `{{ name }}` or `{{name}}`. Why did you add only trailing space?

Comment (fortmarek) [reply Templates/AppProject.stencil:25]: yep, you are right. The space does not matter but it looks weird, so I'll delete it 🙂

**Review by laxmorek — APPROVED** [review]

**Review by kwridan — APPROVED** [review]
